Source/WebCore/css/calc/CSSCalcOperationNode.cpp

```diff
@@ -1369,6 +1370,14 @@ double CSSCalcOperationNode::evaluateOperator(CalcOperator op, const Vector<doub
     case CalcOperator::Tan: {
         if (children.size() != 1)
             return std::numeric_limits<double>::quiet_NaN();
+        double x = std::fmod(children[0], piDouble * 2);
+        // std::fmod can return negative values.
+        x = x < 0 ? piDouble * 2 + x : x;
+        ASSERT(x >= 0 && x <= piDouble * 2 || std::isnan(x));
```

Comment:
Could write this:
```
    ASSERT(!(x < 0));
    ASSERT(!(x > piDouble * 2));
```
I like separate assertions so you can tell which one fired instead of `&&`. I also prefer handling `NaN` without an explicit check by taking advantage of how its behavior is defined. But not sure you prefer either of these.

Reply:
Sure I can write this if it handles NaN correctly.